api/docs/feature-development.md

```diff
@@ -0,0 +1,238 @@
+# Feature Development in Rails
+
+## Creating a Model
+
+1.  Take a look at the erd first by running `rails erd`.
+2.  Once you think about the data model you need to create, make a test in `/specs/models`.
+
+```rb
+# frozen_string_literal: true
+
+ require 'rails_helper'
+
+ RSpec.describe NewModel, type: :model do
+ end
+
+```
+
+then, run `rails spec` and see it fail, you only should ever write just enough code to make your expectation of what should happen pass.
+
+Obviously, we can go and create the model now, because our intent is to have the model exist.
+
+Only add columns for the model **after** you've added a test for the column/association.
+
+## Creating a resolver in GraphQL
+
+In GraphQL a resolver can be one of 2 types, a `query`, or a `mutation`.
+There are a few oddities around GraphQL in Ruby, but for the most part things will translate 1 to 1 in terms of how GraphQL works in other platforms.
+
+### Creating a Resolver
+
+When you're creating a resolver there are a few things you should consider before creating one. **CRUD** (Create, Read, Update, Delete)
```

Comment:
Am I supposed to consider the CRUD operations? Or something else. Small nitpick.

Reply:
@ajtoo idk, I just wanted to drive the idea home because we clearly don't follow it in our codebase, currently.

Reply:
LOL I totally understand. May I suggest

"When creating a resolver, consider which type of **CRUD** (create, read, update, delete) operation you want . You typically want to keep these operations separate for clarity."